app/jobs/scheduled/clean_up_inactive_users.rb

```diff
@@ -0,0 +1,29 @@
+module Jobs
+
+  class CleanUpInactiveUsers < Jobs::Scheduled
+    every 1.day
+
+    def execute(args)
+      if SiteSetting.inactive_time_before_cleaning > 0
+        destroyer = UserDestroyer.new(Discourse.system_user)
+
+        User.joins("LEFT JOIN posts ON posts.user_id = users.id")
+          .where(last_posted_at: nil)
+          .where("posts.user_id IS NULL")
+          .where("users.last_seen_at < ?", SiteSetting.inactive_time_before_cleaning.months.ago)
```

Comment:
I think days is fine here. Name is hard, what about something like `clean_up_inactive_users_after_days`? (cc @eviltrout)